starklings-backend/starklings_backend/routes.py

```diff
@@ -0,0 +1,95 @@
+from flask import request, Blueprint
+import asyncio
+import bcrypt
+from sqlalchemy.exc import IntegrityError
+from flask_sqlalchemy import SQLAlchemy
+from starklings_backend.utils import verify_email
+from starklings_backend.models.shared import db
+from starklings_backend.models.user import Starklingsuser
+from starklings_backend.exercise import verify_exercise
+from checker import ExerciceFailed
+import tempfile
+
+app_routes = Blueprint('app_routes', __name__)
+
+@app_routes.route('/', methods=['GET'])
+def landing():
+    return 'Starklings API'
+
+#######################
+#     Users Routes    #
+#######################
+@app_routes.route('/registerUser', methods=['POST'])
+def register_user():
+    """
+    Inserts a new user in the Database
+    @TODO: Starknet ID / Signature and implements model
+    """
+    try:
+        signature = request.json.get('signature', None)
+        wallet_address = request.json.get('wallet_address', None)
+        username = request.json.get('username', wallet_address)
+        if None in [wallet_address, signature]:
+            return "Wrong form", 400 
+        #@TODO: Check Signature validity
+        
+        user = Starklingsuser(wallet_address=wallet_address, signature=signature, username=username)
+        db.session.commit()
+        return f'Welcome! {username}', 200
+
+    except IntegrityError as e:
+        db.session.rollback()
+        return 'User Already Exists', 400
+    except AttributeError:
+        return 'Provide an Email and Password in JSON format in the request body', 400
+
+
+@app_routes.route('/fetchUserInfo', methods=['POST'])
+def fetch_user_info():
+    """
+    Authenticate a user
+    @TODO Implements Fetch User Information
+    """
+    try:
+        wallet_address = request.json.get('wallet_address', None)
+        if not wallet_address:
+            return 'Missing address', 400
+        user = Starklingsuser.query.filter_by(wallet_address=wallet_address).first()
+        if not user:
+            return 'User Not Found!', 404
+        
+        return f'Logged in, Welcome {user.username}!', 200
+    except AttributeError as e:
+        print(e)
+        return 'Provide the wallet address in JSON format in the request body', 400
+
+
+#######################
+#   Exercises Routes  #
+#######################
+@app_routes.route('/exercise', methods=['POST'])
```

Comment:
The route is not very expressive, wdyt about `/exercises/check`?

Reply:
Yeah i'll change to that, correct we could have other use case later on exercises